src/lang/term/term_cache.ml

```diff
@@ -80,58 +26,9 @@ let retrieve ?name ~trim parsed_term : Term.t option =
       | Some name -> Startup.time (Printf.sprintf "%s cache retrieval" name) fn
   in
   report (fun () ->
-      try
-        match cache_filename ?name ~trim parsed_term with
-          | None -> None
-          | Some filename ->
-              if Sys.file_exists filename then (
-                let ic = open_in_bin filename in
-                Fun.protect
-                  ~finally:(fun () -> close_in_noerr ic)
-                  (fun () ->
-                    let term = Marshal.from_channel ic in
-                    (match name with
-                      | Some name ->
-                          Startup.message "Loading %s from cache!" name
-                      | None -> ());
-                    Some term))
-              else None
-      with
-        | Failure msg
-          when String.starts_with ~prefix:"input_value: unknown code module" msg
-          ->
-            (match name with
-              | Some name ->
-                  Startup.message
-                    "Liquidsoap binary changed: %s cache invalidated!" name
-              | None -> ());
-            None
-        | exn ->
-            let bt = Printexc.get_backtrace () in
-            let exn = Printexc.to_string exn in
-            if Sys.getenv_opt "LIQ_DEBUG_CACHE" <> None then
-              Startup.message "Error while loading cache: %s\n%s" exn bt
-            else Startup.message "Error while loading cache: %s" exn;
-            None)
+      if Cache.enabled () then
+        Cache.retrieve ?name (cache_filename ?name ~trim parsed_term)
+      else None)
```

Comment:
Should move `report` inside the enabled branch so we don't report when cache is disabled.